frontend/packages/console-shared/src/hooks/flag.ts

```diff
@@ -1,9 +1,14 @@
 // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
 // @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
+import { UseFlag, UseSetFlag } from '@console/dynamic-plugin-sdk';
+import { setFlag } from '@console/internal/actions/features';
 import { RootState } from '@console/internal/redux';
 
 export const useFlag: UseFlag = (flag) =>
   useSelector<RootState, boolean>(({ FLAGS }) => FLAGS.get(flag));
 
-type UseFlag = (flag: string) => boolean;
+export const useSetFlag: UseSetFlag = () => {
+  const dispatch = useDispatch();
+  return (flag: string, value: boolean) => dispatch(setFlag(flag, value));
```

Comment:
this needs to be wrapped in useCallback to make sure we do not return new function on every render